common/autoware_agnocast_wrapper/include/autoware/agnocast_wrapper/autoware_agnocast_wrapper.hpp

```diff
@@ -410,17 +431,17 @@ class ROS2Publisher : public Publisher<MessageT>
 
   AUTOWARE_MESSAGE_SHARED_PTR(MessageT) allocate_output_message_shared() override
   {
-    return AUTOWARE_MESSAGE_SHARED_PTR(MessageT){std::make_unique<MessageT>()};
+    return AUTOWARE_MESSAGE_SHARED_PTR(MessageT){std::make_shared<MessageT>()};
   }
 
   void publish(AUTOWARE_MESSAGE_UNIQUE_PTR(MessageT) && message) override
   {
-    publisher_->publish(std::move(message)->move_ros2_ptr());
+    publisher_->publish(std::move(message).move_ros2_ptr());
   }
 
   void publish(AUTOWARE_MESSAGE_SHARED_PTR(MessageT) && message) override
   {
-    publisher_->publish(std::move(message)->move_ros2_ptr());
+    publisher_->publish(*message);
```

Comment:
I think `std::move(message).move_ros2_ptr()` would be better, as it is consistent with other parts.

Reply:
Understood. Let's use `*message` then.